src/utils/deploy.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Interfaces, Flags } from '@oclif/core';
+import { ConfigAggregator, SfdxPropertyKeys } from '@salesforce/core';
+import { Duration } from '@salesforce/kit';
+import { ComponentSet, ComponentSetBuilder, DeployResult, MetadataApiDeploy } from '@salesforce/source-deploy-retrieve';
+import { getPackageDirs, getSourceApiVersion } from './project';
+import { API, TestLevel, TestResults } from './types';
+
+type Options = {
+  api: API;
+  'target-org': string;
+  'test-level': TestLevel;
+  'api-version'?: string;
+  'dry-run'?: boolean;
+  'ignore-errors'?: boolean;
+  'ignore-warnings'?: boolean;
+  manifest?: string;
+  metadata?: string[];
+  'source-dir'?: string[];
+  tests?: string[];
+  wait?: Duration;
+  verbose?: boolean;
+};
+
+export function resolveRestDeploy(): API {
+  const restDeployConfig = ConfigAggregator.getValue(SfdxPropertyKeys.REST_DEPLOY).value;
+
+  if (restDeployConfig === 'false') {
+    return API.SOAP;
+  } else if (restDeployConfig === 'true') {
+    return API.REST;
+  } else {
+    return API.SOAP;
+  }
+}
+
+export async function executeDeploy(
+  opts: Partial<Options>
+): Promise<{ deploy: MetadataApiDeploy; componentSet: ComponentSet }> {
+  const componentSet = await ComponentSetBuilder.build({
+    apiversion: opts['api-version'],
+    sourceapiversion: await getSourceApiVersion(),
+    sourcepath: opts['source-dir'],
+    manifest: opts.manifest && {
+      manifestPath: opts.manifest,
+      directoryPaths: await getPackageDirs(),
+    },
+    metadata: opts.metadata && {
+      metadataEntries: opts.metadata,
+      directoryPaths: await getPackageDirs(),
+    },
+  });
+
+  const deploy = await componentSet.deploy({
+    usernameOrConnection: opts['target-org'],
+    apiOptions: {
+      checkOnly: opts['dry-run'] || false,
+      ignoreWarnings: opts['ignore-warnings'] || false,
+      rest: opts.api === API.REST,
+      rollbackOnError: !opts['ignore-errors'] || false,
+      runTests: opts.tests || [],
+      testLevel: opts['test-level'],
+    },
+  });
+
+  return { deploy, componentSet };
+}
+
+export const apiFlag = (opts: Partial<Interfaces.OptionFlag<API>> = {}): Interfaces.OptionFlag<API> => {
+  return Flags.build<API>({
+    options: Object.values(API),
+    helpValue: `<${Object.values(API).join('|')}>`,
+    defaultHelp: async (): Promise<API> => Promise.resolve(resolveRestDeploy()),
```

Comment:
can you try this without the defaultHelp?  If I remember what I wrote, it *should* put the `default` in the help and only require `defaultHelp` for places where the help needs to be different (typically because it's not type string)